services/src/main/java/org/keycloak/protocol/oidc/DefaultTokenExchangeProvider.java

```diff
@@ -603,6 +604,13 @@ protected UserModel importUserFromExternalIdentity(BrokeredIdentityContext conte
                 logger.debugf("Email verified automatically after registration of user '%s' through Identity provider '%s' ", user.getUsername(), context.getIdpConfig().getAlias());
                 user.setEmailVerified(true);
             }
+
+            event.clone()
+                    .event(EventType.REGISTER)
+                    .user(user.getId())
+                    .detail(Details.REGISTER_METHOD, "token-exchange")
```

Comment:
@douglaspalmer What about adding the broker alias as a detail? It should help to better recognize the broker from which the user originated.